src/Http/Livewire/Overlays/BrowseArticles.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Astrogoat\Blog\Http\Livewire\Overlays;
+
+use Astrogoat\Blog\Models\Article;
+use Helix\Lego\Http\Livewire\Modal;
+
+class BrowseArticles extends Modal
+{
+    public string $articleInputId;
+
+    public string $query = '';
+
+    public $articles = [];
+
+    public function mount($articleInputId)
+    {
+        $this->articleInputId = $articleInputId;
+    }
+
+    public function updatedQuery($query)
+    {
+        if (! $query) {
+            return $this->articles = [];
+        }
+
+        $this->articles = Article::where(Article::getDisplayKeyName(), 'LIKE', "%{$query}%")
```

Comment:
Curious why these are functions `getDisplayKeyName` and `getAuthorName`? It Seems like an unnecessary extra step, since the strings are hard coded in the function.

Reply:
The `getDisplayKeyName` is an abstraction need for Strata to work.